pkg/operator/encryption/key_controller_encryption.go

```diff
@@ -181,6 +181,7 @@ func (c *encryptionKeyController) generateKeySecret(gr schema.GroupResource, key
 				encryptionSecretInternalReason: internalReason,
 				encryptionSecretExternalReason: externalReason,
 			},
+			Finalizers: []string{encryptionSecretFinalizer},
```

Comment:
Why would we want to do that?

Reply:
Secrets in that state are read-only and can't be modified. I'm thinking about how would `pod_state` controller react to a read key in a state like that? Would it deadlock because it couldn't annotate the key?

Reply:
> Secrets in that state are read-only and can't be modified. I'm thinking about how would `pod_state` controller react to a read key in a state like that? Would it deadlock because it couldn't annotate the key?

I think the read-only state is only for namespaces.

I ran:

```
$ oc delete secret --all -n openshift-config-managed
```

in the middle of a rotation.  None of the secrets were deleted, but they did gain a deletion timestamp.

Everything kept working _and_ the controllers were able to keep updating the secrets.